Update RAD Manager to support new features in MVP3 (DCP268836, DCP271577 / RN42915)

tobedeprez opened the pull request.

VeerleLedoux commented on RADWidgets/Compatibility.cs:
```diff
@@ -0,0 +1,48 @@
+		[MethodImpl(MethodImplOptions.NoInlining)]
+		private static void CheckAddRADSharedModelGroupMessage(IEngine engine)
+		{
+			Type type = typeof(AddRADSharedModelGroupMessage);
```
Guess we can use the ADDRADSubgroupMessage in the next version

tobedeprez replied:
Not applicable anymore after change to RAD Toolkit

VeerleLedoux commented on RADDataSources/AnomalyScoreCache.cs:
```diff
@@ -74,6 +86,26 @@ private void UpdateAnomalyScoreData(int dataMinerID, string groupName, DateTime
+		{
+			try
+			{
+				if (subGroupID != Guid.Empty && subGroupID != null)
```
Guid cannot be null

tobedeprez replied:
Done

VeerleLedoux commented on RADWidgets/RadSubgroupEditor.cs:
```diff
@@ -0,0 +1,196 @@
+			_otherSubgroups = allSubgroups;
+			if (settings != null)
+				_otherSubgroups = _otherSubgroups.Where(s => s.ID == settings.ID).ToList();
+
```
just double-checking: this fills otherSubgroups with a list containing only the subgroup with the same ID as settings? It is not that the intended behavior was to filter out the current subgroup from otherSubgroups

tobedeprez replied:
Yes, that was the intended behavior. Changed now

VeerleLedoux commented on RadUtils/RadMessageHelper.cs:
```diff
@@ -147,6 +365,204 @@ private static RetrainMADModelResponseMessage RetrainParameterGroup(
+					IsMonitored = parameterGroupInfoResponse.IsMonitored,
+				};
+			}
+			else if (response is GetRADSharedModelGroupInfoResponseMessage sharedModelGroupInfoResponse)
```
As you know, some changes are coming in the API ;-)

tobedeprez commented on RADWidgets/RadLabelEditorDialog.cs:
```diff
@@ -0,0 +1,156 @@
+				{
+					_detailsLabel.Text = $"The labels {duplicateLabels.Select(g => g.Key).HumanReadableJoin()} are duplicated.";
+					_detailsLabel.IsVisible = true;
+					_okButton.Tooltip = "Set the parameter labels above. Note that before submitting the group, you will have to remove the duplicated labels.";
```
Done

tobedeprez commented:
1. Will be tacked in DCP270000
2. As discussed, it looks like you're unselecting all parameters here, in which case there will indeed not be a trend graph.
3. The parameter dropdown should now light up in red when an element has no parameters. The same now also happens when selecting 'Add group for each element with given connector' and for the parameter selectors in the subgroup editor dialogs in 'Add group with shared model'.
4. Same, will be tackled by DCP270000. Not sure I'll be able to fix the 'vector too long' one. Can you reproduce that one, it seems like it more a bug in SLAnalytics or somewhere in SLAnalytics where we should return a more user-friendly exception?
5. It looks like the group has no name. Not sure how you were able to add that one? Normally, you shouldn't be able to
6. (no 6)
7. Changed
8. This was a bug in SLAnalytics, but we decided to no longer block different groups having the same parameters (DCP273218).
9. Should not happen anymore.
10. Shared model group is disabled by default now
11. Selecting only specific elements is DCP269337. I've added a button 'More details...' that shows more details.
12. As discussed, kept the behavior.
13. They should be blocked now.
14. The subgroups do not really have an order. I now sorted them alphabetically in this representation.
15.
 - (first 15): For now this is OK, I think. It is linked with a discussion we had during review meeting, which we will have to discuss at some point
 - (second 15) Similar to 12
16. As discussed. Changed it to a dropdown instead
17. (no 17)
18. There is a 15 minute time-out in interactive automation scripts. There isn't really anything I can do about that as far as I'm aware. If it is really annoying, we can reach out to Automation & Orchestration to see whether something can be done.

tobedeprez commented on RADWidgets/Widgets/Editors/RadSubgroupEditor.cs:
```diff
@@ -31,7 +31,7 @@ public RadSubgroupEditor(IEngine engine, List<RadSubgroupSelectorItem> allSubgro
 
-			_parameterSelectors = new List<Tuple<Label, ParameterInstanceSelector>>(parameterLabels.Count);
+			_parameterSelectors = new List<Tuple<Label, ParameterInstanceSelector>>(_parameterLabels.Count);
 			for (int i = 0; i < parameterLabels.Count; i++)
```
Done

AdelinaSpatariu commented:
1. Label column is added for single groups as well. Do we want that?
2. Normally we do not allow subgroups with the same name but if I try to do it and add a space after one of the subgroups name then I end up with 2 subgroups with the same name :) 
3. Is there a reason why you write the group name with capital letters in the Edit window?

tobedeprez commented:
1. There is not really an easy way to remove it, and also you can make a single group with labels if you really want (but you'll have to use the shared model option in add groups). This might also become better once we get a better idea of how exactly we do shared vs. non-shared model groups (cfr. point 15 above)
2. Trailing and leading spaces are now removed from group or subgroup names
3. As discussed, removed the subgroup name alltogether (it is shown in the dropdown itself)

AdelinaSpatariu commented:
I do not have other remarks for now
